filetracker/servers/storage.py

```diff
@@ -159,19 +160,70 @@ def store(self, name, data, version, size=0, compressed=False, digest=None):
                 if temp_file_path and os.path.exists(temp_file_path):
                     os.unlink(temp_file_path)
 
-            if os.path.exists(link_path):
+            if _path_exists(link_path):
                 # Lend the link lock to delete().
                 # Note that DB lock has to be released in advance, otherwise
                 # deadlock is possible in concurrent scenarios.
                 self.delete(name, version, lock=False)
 
             _create_file_dirs(link_path)
-            os.symlink(blob_path, link_path)
+            rel_blob_path = os.path.relpath(blob_path,
+                                            os.path.dirname(link_path))
+            os.symlink(rel_blob_path, link_path)
 
-            os.utime(link_path, (version, version))
+            lutime(link_path, version)
+            return version
 
     def delete(self, name, version, lock=True):
-        pass
+        """Removes a file from the storage.
+
+                   Args:
+                        name: name of the file being deleted.
+                            May contain slashes that are treated as path separators.
+                        version: file "version" that is meant to be deleted
+                            If the file that is stored has newer version than provided,
+                            it will not be deleted.
+                        lock: whether or not to acquire locks
+                            This is for internal use only,
+                            normal users should always leave it set to True.
+                   Returns whether or not the file has been removed
+                   or None if no such file exists.
+                """
+        link_path = self._link_path(name)
+        if not _path_exists(link_path):
+            return None
+        if _file_version(link_path) > version:
```

Comment:
Possible race condition: the file may have been already deleted, should probably lock earlier.